ee/WFCORE-4135_Support_for_ read_only_server_config_dir.adoc

```diff
@@ -0,0 +1,99 @@
+
+The goal of this feature is to make `--read-only-server-config=<server config file name>` setting work in standalone mode even if the whole configuration directory pointed by `jboss.server.config` is a read only directory.
+
+Domain mode changes are not initially included on this feature.
+
+== Required changes for cloud images
+
+The cloud images should be smart enough to detect if the server configuration directory is a read only directory, and in this case, the launch command for the underlying server will incorporate the `--read-only-server-config=<server config file name>` argument to make it possible to start the server under this circumstance.
+
+That means, it is not required to take any action if the cloud image configuration detects a read-only server configuration directory or file, however, a log info trace will be added to the standard output to inform the user the server is working with a read-only server configuration directory.
+
+Users will always be able to force the activation/deactivation of the `--read-only-server-config=<server config file name>` setting by using `IS_READ_ONLY_SERVER_CONFIG=<true|false>` environment variable. The use of the environment variable prevails over the automatic detection done by the scripts.
+
+Server configuration file marker replacements on a read only configuration file will not be supported by this feature. That means, users, who are relying on marker replacements over a user-supplied server configuration file stored in a read-only directory, should adapt their configuration to do not use marker replacements. We could adapt the server behavior for this scenario, but it sounds useless; marker replacements have never been a supported way to configure the server.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/EAP7-1119[EAP7-1119]
+* https://issues.jboss.org/browse/WFCORE-4135[WFCORE-4135]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFCORE-3937[WFCORE-3937]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+
+[X] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+WildFly Core
+WildFly Cekit Modules
+
+== Requirements
+
+=== Hard Requirements
+
+* The `--read-only-server-config=<server config file name>` setting will be expanded to allow, not only a read only configuration file, but also a read only configuration directory.
+* The user can activate/deactivate this setting by using the following environment variable `IS_READ_ONLY_SERVER_CONFIG=<true|false>`
```

Comment:
I expect the last three points are just for OCP environments.

Reply:
Yes, you're right. Done.